pkg/runtime/reconciler.go

```diff
@@ -401,13 +465,13 @@ func (r *resourceReconciler) setResourceManaged(
 	}
 	orig := res.RuntimeObject().DeepCopyObject()
 	r.rd.MarkManaged(res)
-	rlog.Enter("kc.Patch (all)")
+	rlog.Enter("kc.Patch (metadata + spec)")
 	err = r.kc.Patch(
 		ctx,
 		res.RuntimeObject(),
 		client.MergeFrom(orig),
 	)
```

Comment:
`r.patchResourceMetadataAndSpec` only update object metadata when a changes in annotations is detected. `setResourceManaged`/ `MarkManaged` uses `ObjectMetadata.SetFinalizers` which is updating a seperate field (`Finalizers)`
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/controller/controllerutil/controllerutil.go#L351-L360

Reply:
Yeah, There is no functionality change here. Just an update of log statement.

r.kc.Patch() will update finalizers, labels i.e. all of metadata and spec. We are not calling "r.patchResourceMetadataAndSpec" here which currently only updates for annotation difference(I will be updating that).